presto-iceberg/src/main/java/com/facebook/presto/iceberg/util/StatisticsUtil.java

```diff
@@ -13,36 +13,47 @@
  */
 package com.facebook.presto.iceberg.util;
 
+import com.facebook.presto.common.type.FixedWidthType;
 import com.facebook.presto.hive.metastore.HiveColumnStatistics;
 import com.facebook.presto.hive.metastore.PartitionStatistics;
 import com.facebook.presto.iceberg.IcebergColumnHandle;
+import com.facebook.presto.spi.statistics.ColumnStatisticType;
 import com.facebook.presto.spi.statistics.ColumnStatistics;
 import com.facebook.presto.spi.statistics.Estimate;
 import com.facebook.presto.spi.statistics.TableStatistics;
+import com.google.common.base.Joiner;
 import org.apache.iceberg.PartitionSpec;
 
+import java.util.Arrays;
+import java.util.EnumSet;
 import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
-import static com.facebook.presto.iceberg.util.HiveStatisticsMergeStrategy.NONE;
-import static com.facebook.presto.iceberg.util.HiveStatisticsMergeStrategy.USE_NDV;
-import static com.facebook.presto.iceberg.util.HiveStatisticsMergeStrategy.USE_NULLS_FRACTIONS;
-import static com.facebook.presto.iceberg.util.HiveStatisticsMergeStrategy.USE_NULLS_FRACTION_AND_NDV;
+import static com.facebook.presto.spi.statistics.ColumnStatisticType.NUMBER_OF_DISTINCT_VALUES;
 
 public final class StatisticsUtil
 {
     private StatisticsUtil()
     {
     }
 
-    public static TableStatistics mergeHiveStatistics(TableStatistics icebergStatistics, PartitionStatistics hiveStatistics, HiveStatisticsMergeStrategy mergeStrategy, PartitionSpec spec)
+    /**
+     * Attempts to merge statistics from Iceberg and Hive tables.
+     * <br>
+     * Statistics from Hive are only merged if the corresponding flag exists in
+     * the {@code mergeFlags}.
+     */
+    public static TableStatistics mergeHiveStatistics(TableStatistics icebergStatistics, PartitionStatistics hiveStatistics, EnumSet<ColumnStatisticType> mergeFlags, PartitionSpec spec)
     {
-        if (mergeStrategy.equals(NONE) || spec.isPartitioned()) {
+        if (spec.isPartitioned()) {
```

Comment:
Seems currently we could not get valid values of `data_size` and `distinct_values_count` for a partitioned iceberg table configured with hive. Because we do not write and use iceberg statistics when configured with hive, and do not use hive statistics when iceberg table is partitioned.

Maybe we should write and use iceberg statistics when table is partitioned?

Reply:
I have another draft on my local machine which adds support for table statistics in some cases of partitioned tables. I will try to get it submitted early next week, but I think changing that behavior is out of scope for this PR.

Reply:
Sure, looking forward to see it.